### ISSUE TITLE
Finetuning with limited computational resources

### ISSUE THREAD

**Issue by PixarHero** [DeepLearnXMU/LLaVE#9]
Thank you for your excellent work! I would like to inquire about fine-tuning with limited computational resources. Specifically, I have access to 2 A100 GPUs with 80GB of memory each. Given these constraints, I would like to know if fine-tuning is feasible and if you have any advice for optimizing the process. Can using a small batch size lead to poor model performance or result in a loss of zero or is there a minimum batch size threshold below which training fails to work? Thx!

**Comment by Nonoor** [comment]
Given your computational resources (2×A100 GPUs with 80GB of memory), I recommend fine-tuning a smaller model (e.g., 0.5B).

To further optimize for limited resources, consider using LoRA fine-tuning. You can enable it in the training script with the following flags:
`
--lora_enable True 
--lora_r 32 
--lora_alpha 64
`

Please note that when using LoRA, you may need to adjust the learning rate from what you'd typically use in full fine-tuning to achieve optimal performance.

Regarding batch size: if the `per_device_train_batch_size` is too small (e.g., 1 per GPU), the number of negative samples per batch will be very limited (only 2 in total with 2 GPUs). This significantly reduces the likelihood of encountering hard negatives (Each positive example has only one random negative example), which may cause the loss to drop to zero and hinder effective training. In my experience, the global batch size (before counting for `gradient_accumulation_steps`) should be at least 16–32 to ensure stable training.

**Comment by PixarHero** [comment]
Thank you very much for your detailed advice and insights! I will adjust the learning rate and training strategy accordingly to better utilize our hardware.
